compiler/rustc_lint/messages.ftl

```diff
@@ -197,15 +197,16 @@ lint_dangling_pointers_from_locals = {$fn_kind} returns a dangling pointer to dr
     .ret_ty = return type is `{$ret_ty}`
     .local_var = local variable `{$local_var_name}` is dropped at the end of the {$fn_kind}
     .created_at = dangling pointer created here
+    .help_visit = for more information, see <https://doc.rust-lang.org/reference/destructors.html>
     .note = a dangling pointer is safe, but dereferencing one is undefined behavior
 
-lint_dangling_pointers_from_temporaries = a dangling pointer will be produced because the temporary `{$ty}` will be dropped
-    .label_ptr = this pointer will immediately be invalid
-    .label_temporary = this `{$ty}` is deallocated at the end of the statement, bind it to a variable to extend its lifetime
-    .note = pointers do not have a lifetime; when calling `{$callee}` the `{$ty}` will be deallocated at the end of the statement because nothing is referencing it as far as the type system is concerned
-    .help_bind = you must make sure that the variable you bind the `{$ty}` to lives at least as long as the pointer returned by the call to `{$callee}`
-    .help_returned = in particular, if this pointer is returned from the current function, binding the `{$ty}` inside the function will not suffice
+lint_dangling_pointers_from_temporaries = expression creates a dangling pointer to dropped temporary `{$ty}`
```

Comment:
I'm not sure that's the a description of the issue, I feel like it's a bit misleading, the temporary is not yet dropped (otherwise this would insta UB), it's dropped only after creating the pointer, and the dropping of the temporary which makes the pointer dangling.

The previously wording was clearer IMO on this point. Not sure how to best improve that.

Reply:
Hmm, I thought that detail did not matter. Do you have a reference for your claim that that would be insta UB?

Reply:
> Do you have a reference for your claim that that would be insta UB?

For `String::new().DROPPED.as_ptr()` it's apparently a grey area https://github.com/rust-lang/unsafe-code-guidelines/issues/188 where use-after-dropped and use-after-free are different regarding opsem. I was told that it should probably be UB and is heavily discouraged.

But for allocation like `"".to_string().DROPPED.as_ptr()` Miri flags it as UB[^1] on the `as_ptr()` call, so :shrug:

[^1]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=1c4f3446888f2b92d486a5f478722640

Reply:
I created https://github.com/rust-lang/miri/issues/4668 to hopefully clarify this...

Reply:
Okay, new version pushed.